git_history_of_project/git_history_of_project.md

```diff
@@ -0,0 +1,620 @@
+[CC BY-SA 3.0](https://creativecommons.org/licenses/by-sa/3.0/deed.en).
+
+<div class = "options">
+If you have a repository with at least a few commits in it, such as the one created in the previous module in this series, you can use this module as a framework for exploring that instead.
+
+Alternatively, you can clone any repository you find on [GitHub](https://github.com) using
+
+```
+git clone https://www.github.com/RepositoryOwner/RepositoryName
+```
+
+and explore how it was created!
+</div>
+
+## Seeing prior commits
+
+Keeping track of all versions and being able to see and compare them is the entire point of using version control on a project you are working on alone. It is also a huge part of working on a project with others, if you are collaborating in the future.
+
+Each time you `commit` to Git, you are marking the current state of your project as a checkpoint that you can return to. The records of these changes are referred to as [commits](https://swcarpentry.github.io/git-novice/reference.html#commit)
+
+
+You are going to learn about two ways to refer to past commits:
+
+- Using `HEAD` to refer to the latest `commit`.
+- Using the commit number assigned to a particular `commit`.
+
+
+![The Commit Stack: 3 flat white boxes, stacked on top of each other. The top box is labeled "last committed version of repository", the middle box is labeled "next to last version of repository" and the bottom is labeled "previous version of repository". Three dots below the lowest box indicate that this pattern continues.](media/Commit_stack.svg)
+
+### Using HEAD
+
+The `HEAD` refers to the most recently commit to your repository. It won't include any changes that haven't been committed to the repository yet. You can see that most recent commit by entering `git show HEAD`.
+
+```
+$ git show HEAD
+commit 584977a18319f539f3f5ecde6176db2ba0081cd9 (HEAD -> main, origin/main)
+Author: Vlad Dracula <vlad@tran.sylvan.ia>
+Date:   Mon Mar 14 14:59:40 2022 -0400
+
+    Ignore data files and the results folder.
+
+diff --git a/.gitignore b/.gitignore
+new file mode 100644
+index 0000000..f451386
+--- /dev/null
++++ b/.gitignore
+@@ -0,0 +1,2 @@
++*.dat
++results/
+```
+
+The last thing that Dracula committed to the planets repository was the `.gitignore` file in which he told Git to "Ignore data files and the results folder."
+
+<div class = "care">
+Sometimes a command will output many lines of code that may not be relevant to you. For example the `show` command will output not just the commit message, but also every single change that you made in that commit.
+
+The more you see this type of output, the more comfortable you will get with extracting the information you need and ignoring the rest.
+</div>
+
+<div class = "important">
+If the output of a command is longer than the number of lines your console displays, you can navigate that output using the down and up arrows on your keyboard, or press `q` to skip to the end of that output.
```

Comment:
doesn't `q` actually just "quit" you out without moving ahead to the end of the output?